File: Boundary traversal of matrix - GFG/boundary-traversal-of-matrix.cs
//{ Driver Code Starts
//Initial Template for C#

using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode
{

    class GFG
    {
        static void Main(string[] args)
        {
            int testcases;// Taking testcase as input
            testcases = Convert.ToInt32(Console.ReadLine());
            while (testcases-- > 0)// Looping through all testcases
            {
                var stringArray = Console.ReadLine().Split(' ');
                int n = int.Parse(stringArray[0]);
                int m = int.Parse(stringArray[1]);
                stringArray = Console.ReadLine().Split(' ');
                List<List<int>> A = new List<List<int>>();
                for (int i = 0; i < n * m; i += m)
                {
                    List<int> row = new List<int>();
                    for (int j = 0; j < m; j++)
                    {
                        row.Add(int.Parse(stringArray[i + j]));
                    }
                    A.Add(row);
                }
                Solution obj = new Solution();
                var res = obj.boundaryTraversal(A,n,m);
                foreach(int i in res){
                    Console.Write(i + " ");
                }
                Console.WriteLine();
            }
        }
    }

    
// } Driver Code Ends
//User function Template for C#

    class Solution
    {
        //Function to return list of integers that form the boundary 
        //traversal of the matrix in a clockwise manner.
        public List<int> boundaryTraversal(List<List<int>> a, int n, int m)
        {
            List<int> v = new List<int>();
            for (int i = 0; i < m; i++) v.Add(a[0][i]);
            for (int i = 1; i < n; i++) v.Add(a[i][m-1]);
            if (n > 1) for (int i = m-2; i >= 0; i--) v.Add(a[n-1][i]);
            if (m > 1) for (int i = n-2; i > 0; i--) v.Add(a[i][0]);
            return v;
        }
    }
}

//{ Driver Code Starts.
// } Driver Code Ends

File: Boundary traversal of matrix - GFG/boundary-traversal-of-matrix.js
//{ Driver Code Starts
//Initial Template for javascript
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readLine() {
    return inputString[currentLine++];
}

function main() {
    let t = parseInt(readLine());
    for(let i=0;i<t;i++)
    {
        let input_line = readLine().split(' ');
        let n = parseInt(input_line[0]);
        let m = parseInt(input_line[1]);
        
        input_line = readLine().split(' ');
        let matrix = new Array(n);
        for(let i=0;i<n;i++)
        {
            let a = new Array(m);
            for(let j=0;j<m;j++)
            {
                a[j] =parseInt(input_line[i*m+j]);
            }
            matrix[i] = a;
        }
        
        let obj = new Solution();
        let ans = obj.boundaryTraversal(matrix, n, m);
        let s = "";
        for(let i = 0 ; i < ans.length ;i++)
        {
            if(ans[i]==-0)
                ans[i]=0;
            s+=ans[i];
            s+=" ";
        }
        console.log(s);
    }
}
// } Driver Code Ends


//User function Template for javascript

class Solution 
{
    boundaryTraversal(a, n, m)
    {
        let v = [];
        for (let i = 0; i < m; i++) v.push(a[0][i]);
        for (let i = 1; i < n; i++) v.push(a[i][m-1]);
        if (n > 1) for (let i = m-2; i >= 0; i--) v.push(a[n-1][i]);
        if (m > 1) for (let i = n-2; i > 0; i--) v.push(a[i][0]);
        return v;
    }
}